src/platform/tls_openssl.c

```diff
@@ -0,0 +1,3457 @@
+ // ticket keying, and callback functions.
+ // 
+typedef struct CXPLAT_SEC_CONFIG {
+
+    //
+    // SSL context used for establishing TLS connections.
+    //
+    SSL_CTX *SSLCtx;
+
+    //
+    // Pointer to the ticket key configuration for session resumption.
+    //
+    QUIC_TICKET_KEY_CONFIG* TicketKey;
+
+    //
+    // TLS-related callbacks for handling crypto events.
+    //
+    CXPLAT_TLS_CALLBACKS Callbacks;
+
+    //
+    // Credential flags specifying various QUIC credential options.
+    //
+    QUIC_CREDENTIAL_FLAGS Flags;
+
+    //
+    // Flags that specify behavior for TLS credential handling.
+    //
+    CXPLAT_TLS_CREDENTIAL_FLAGS TlsFlags;
+
+} CXPLAT_SEC_CONFIG;
+
+//
+// @struct CXPLAT_TLS
+// @brief Represents the state and configuration of a TLS session.
+//
+// This structure holds information necessary to manage a TLS handshake,
+// encryption state, and associated connection details for QUIC.
+//
+typedef struct CXPLAT_TLS {
+
+    //
+    // Pointer to the security configuration used for the TLS session.
+    //
+    CXPLAT_SEC_CONFIG* SecConfig;
+
+    //
+    // Pointer to HKDF label definitions used in the key derivation process.
+    //
+    const QUIC_HKDF_LABELS* HkdfLabels;
+
+    //
+    // Indicates if the endpoint is acting as a server.
+    //
+    BOOLEAN IsServer : 1;
+
+    //
+    // Indicates if a peer certificate has been received.
+    //
+    BOOLEAN PeerCertReceived : 1;
+
+    //
+    // Indicates if the peer's transport parameters have been received.
+    //
+    BOOLEAN PeerTPReceived : 1;
+
+    //
+    // QUIC transport parameter extension type used in the session.
+    //
+    uint16_t QuicTpExtType;
+
+    //
+    // Length of the ALPN buffer.
+    //
+    uint16_t AlpnBufferLength;
+
+    //
+    // Pointer to the ALPN buffer data.
+    //
+    const uint8_t* AlpnBuffer;
+
+    //
+    // Pointer to the Server Name Indication (SNI) string.
+    //
+    const char* SNI;
+
+    //
+    // OpenSSL SSL object used for the TLS handshake and encryption.
+    //
+    SSL *Ssl;
+
+    //
+    // Pointer to internal TLS processing state.
+    //
+    CXPLAT_TLS_PROCESS_STATE* State;
+
+    //
+    // Flags indicating the results of TLS processing.
+    //
+    CXPLAT_TLS_RESULT_FLAGS ResultFlags;
+
+    //
+    // Pointer to the QUIC connection associated with this TLS session.
+    //
+    QUIC_CONNECTION* Connection;
+
+    //
+    // Pointer to derived TLS secrets for encryption and decryption.
+    //
+    QUIC_TLS_SECRETS* TlsSecrets;
+
+} CXPLAT_TLS;
+
+//
+// @struct RECORD_ENTRY
+// @brief Represents a buffered SSL record in a linked list.
+//
+// This structure is used to store an SSL record along with its
+// metAData and linkage in a list. It supports tracking incomplete
+// records and whether the memory should be freed.
+//
+typedef struct RECORD_ENTRY {
+    //
+    // Linked list node for linking Entries in a list.
+    //
+    CXPLAT_LIST_ENTRY Entries;
```

Comment:
This should be `Link` or `Entry` since it's just a link into the list, not the whole list (for which we use plural).